domehunter/__init__.py

```diff
@@ -86,70 +230,291 @@ def is_home(self):
+            if delta_az > 0:
+                delta_az -= 360
+            else:
+                delta_az += 360
+
+        # if updated delta_az is positive, direction is CW
+        if delta_az > 0:
+            # converted delta_az to equivilant in encoder ticks
+            ticks = self._az_to_ticks(delta_az)
+            target_position = self.encoder_count + ticks
+            self._move_cw()
+            # wait until encoder count matches desired delta az
+            while self.encoder_count < target_position:
+                if self.testing:
+                    # if testing simulate a tick for every cycle of while loop
+                    self._simulate_ticks(num_ticks=1)
+                pass
+            self._stop_moving()
+            # compare original count to current just in case we got more ticks
+            # than we asked for
+            old_encoder_count = target_position - ticks
+            # update dome_az based on the actual number of ticks counted
+            self.dome_az += self._ticks_to_az(
+                self.encoder_count - old_encoder_count)
+            # take mod360 of dome_az to keep 0 <= dome_az < 360
+            self.dome_az %= 360
+            # update encoder_count to match the dome_az
+            self.encoder_count = self._az_to_ticks(self.dome_az)
+            pass
+
+        # if updated delta_az is negative, direction is CCW
+        if delta_az < 0:
+            # converted delta_az to equivilant in encoder ticks
+            ticks = self._az_to_ticks(delta_az)
+            target_position = self.encoder_count + ticks
+            self._move_ccw()
+            # wait until encoder count matches desired delta az
+            while self.encoder_count >= target_position:
+                if self.testing:
+                    # if testing simulate a tick for every cycle of while loop
+                    self._simulate_ticks(num_ticks=1)
+                pass
+            self._stop_moving()
+            # compare original count to current, just in case we got more ticks
+            # than we asked for
+            old_encoder_count = target_position - ticks
+            # update dome_az based on the actual number of ticks counted
+            self.dome_az += self._ticks_to_az(
+                self.encoder_count - old_encoder_count)
+            # take mod360 of dome_az to keep 0 <= dome_az < 360
+            self.dome_az %= 360
+            # update encoder_count to match the dome_az
+            self.encoder_count = self._az_to_ticks(self.dome_az)
+            pass
         pass
 
-    def start_daemon(self):
-        """Maybe start the RCP daemon here?."""
-        raise NotImplementedError
+    def calibrate(self, num_cal_rotations=2):
+        """Calibrate the encoder (determine degrees per tick)"""
+        # rotate the dome until we hit home, to give reference point
+        self._move_cw()
+        self.home_sensor.wait_for_active(timeout=self.wait_timeout)
```

Comment:
why do you need a delay? What if its already on home? Or right next to it? Is that OK? Fine to leave as-is, just posing question.

Reply:
Ah, I see, is it so the testing mode triggers some time later? If so, suggest building in a `time.sleep(1)` within the `if self.testing: statement instead. If normal operations do not need a delay?